runtime/compiler/control/J9Options.cpp

```diff
@@ -1989,11 +1989,15 @@ J9::Options::fePreProcess(void * base)
          std::random_device rd;
          std::mt19937_64 rng(rd());
          std::uniform_int_distribution<uint64_t> dist;
-         compInfo->getPersistentInfo()->setClientUID(dist(rng));
+         uint64_t clientUID = dist(rng);
+         compInfo->getPersistentInfo()->setClientUID(clientUID);
+         jitConfig->clientUID = clientUID;
```

Comment:
We can eliminate `jitConfig->isClient` if we use the convention that a client can never have UID==0. Thus in here we need another check when generating the clientUID. If the random number generator returned 0, we have to call it again.

This also means that we need to initialize jitConfig->clientUID with 0 somewhere very early.